lib/models/BucketPolicy.ts

```diff
@@ -48,64 +48,77 @@ const objectActions = [
     's3:PutObjectTagging',
 ];
 
+export type BucketPolicyMetadata = {
+    Id: string;
+    Version: string;
+    Statement: {
+        Sid: string;
+        Effect: string;
+        Principal: string | { AWS: string[] };
+        Action: string | string[];
+        Resource: string | string[];
+        Condition?: { [key: string]: string | string[] };
```

Comment:
There is a missing intermediate object and value can be number
```suggestion
        Condition?: Record<string, Record<string, string | string[] | number>>;
```

Example 

```json

{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Deny",
            "Principal": "*",
            "Action": "s3:PutObject",
            "Resource": [
                "arn:aws:s3:::amzn-s3-demo-bucket1",
                "arn:aws:s3:::amzn-s3-demo-bucket1/*"
            ],
            "Condition": {
                "NumericLessThan": {
                    "s3:TlsVersion": 1.2
                }
            }
        }
    ]
}
```